internal/store/serverservice.go

```diff
@@ -165,45 +165,41 @@ func (s *Serverservice) AssetByID(ctx context.Context, id string) (*model.Asset,
 	asset.BmcUsername = credential.Username
 	asset.BmcPassword = credential.Password
 
-	// query attributes
-	attributes, _, err := s.client.ListAttributes(ctx, deviceUUID, nil)
+	// query the server object
+	srv, _, err := s.client.Get(ctx, deviceUUID)
 	if err != nil {
 		return nil, errors.Wrap(ErrServerserviceQuery, err.Error())
 	}
 
+	asset.FacilityCode = srv.FacilityCode
+
 	// set bmc address
-	asset.BmcAddress, err = s.bmcAddressFromAttributes(attributes)
+	asset.BmcAddress, err = s.bmcAddressFromAttributes(srv.Attributes)
 	if err != nil {
 		return nil, err
 	}
 
 	// set device state attribute
-	asset.State, err = s.assetStateAttribute(attributes)
+	asset.State, err = s.assetStateAttribute(srv.Attributes)
 	if err != nil {
 		return nil, err
 	}
 
 	// set asset vendor attributes
-	asset.Vendor, asset.Model, asset.Serial, err = s.vendorModelFromAttributes(attributes)
+	asset.Vendor, asset.Model, asset.Serial, err = s.vendorModelFromAttributes(srv.Attributes)
 	if err != nil {
 		return nil, errors.Wrap(ErrVendorModelAttributes, err.Error())
 	}
 
-	// query asset component inventory
-	components, _, err := s.client.GetComponents(ctx, deviceUUID, nil)
```

Comment:
The `GetComponents()` call would have to stay, it returns the component data along with the firmware versions required by flasher,
those (versioned) attributes are not returned by `Get()` on the server, so I suggest keeping this call.

Reply:
Fixed.